### ISSUE TITLE
Xcode 15 beta 6: 'accessor' macro cannot be attached to parameter

### ISSUE THREAD

**Issue by zhigang1992** [joshuawright11/papyrus#13]
<img width="1512" alt="image" src="https://github.com/joshuawright11/papyrus/assets/1057756/6e022603-782e-4bbd-a510-ee5d16edc3b4">

Seems like this is no longer possible?

**Comment by joshuawright11** [comment]
Thanks for the report @zhigang1992.

Tracking on the forums here: https://forums.swift.org/t/accessor-macro-cannot-be-attached-to-a-parameter/66669/2

**Comment by joshuawright11** [comment]
Sorry for the slow followup @zhigang1992, but a fix has now been merged in #14.

Long story short - support macros on function parameters have been removed with no immediate plans to bring them back.

Since they were purely used to decorate function parameter, I replaced them with generic typealiases:

1. `@Path path: String` -> `path: Path<String>`
2. `@Body body: String` -> `body: Body<String>`
3. `@Field field: String` -> `field: Field<String>`
4. `@Header header: String` -> `header: Header<String>`
5. `@Query query: String` -> `query: Query<String>`

All other macros are still working and left the same.

For convenience - I added some inference logic so that users can annotate the function parameters less. Previously, all un-annotated parameters were inferred to be fields.

Now, un-annotated parameters are inferred using the following logic:

1. If it matches a parameter in the path, treat it as `Path<...>`
2. If it's on a `GET`, `HEAD`, or `DELETE` request, treat it as a `Query<...>`
3. Otherwise, treat it as a `Field<...>`

i.e.

```swift
@GET("/user/:id/todos")
func getTodo(id: Int, isComplete: Bool) -> Todo

// is inferred to be

@GET("/user/:id/todos")
func getTodo(id: Path<Int>, isComplete: Query<Bool>) -> Todo
```

and

```swift
@POST("/user/:id/todos")
func createTodo(id: Int, name: String) -> Todo

// is inferred to be

@POST("/user/:id/todos")
func createTodo(id: Path<Int>, name: Field<String>) -> Todo
```
